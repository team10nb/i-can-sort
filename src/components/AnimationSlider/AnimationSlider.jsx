import React from "react";
import { makeStyles, withStyles } from "@material-ui/core/styles";
import Slider from "@material-ui/core/Slider";

const PrettoSlider = withStyles({
    root: {
        color: "#52af77",
        height: 4,
        alignContent: "center",
    },
    thumb: {
        height: 24,
        width: 24,
        backgroundColor: "#fff",
        border: "2px solid currentColor",
        marginTop: -8,
        marginLeft: -12,
        "&:focus, &:hover, &$active": {
            boxShadow: "inherit",
        },
    },
    active: {},
    valueLabel: {
        left: "calc(-50% + 4px)",
    },
    track: {
        height: 5,
        borderRadius: 2,
    },
    rail: {
        height: 5,
        borderRadius: 2,
    },
    mark: {
        height: 5,
    }
})(Slider);

export default function DiscreteSlider(props) {
    const { width, step, max, handleChange, value } = props;

    const useStyles = makeStyles({
        root: {
            width: "500px",
            marginTop: "10px",
        },
    });
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <PrettoSlider
                value={typeof value === "number" ? value : 0}
                aria-labelledby='Progress'
                valueLabelDisplay='auto'
                step={step}
                marks
                min={0}
                max={max}
                onChange={handleChange}
            />
        </div>
    );
}
