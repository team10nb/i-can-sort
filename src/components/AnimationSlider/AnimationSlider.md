### **Developer**

Ruizi Han, Shiliang Chen

Reference: MATERIAL-UI (https://material-ui.com)

### **Introduction**

A progress bar with a slider for user to control the progress of the animation by dragging the slider.

### **Import**

```html
import { makeStyles, withStyles } from "@material-ui/core/styles"; 
import Slider from "@material-ui/core/Slider";
```

### **Props**

| Name         | Type   | Default | Description                                        |
| ------------ | ------ | ------- | -------------------------------------------------- |
| handleChange | fn     |         | Handle change of the slider by user event.         |
| step         | number | 0       | Indicate how length of each step. |
| max          | number | 0       | Indicate the maximum number of step.               |
| value        | number | 0       | Indicate the current value of step.                |
| width        | number | 0       | Indicate the width of the progress bar.            |

### **Demo**

```jsx
<AnimationSlider step={1} max={15} value={5} width={800} />
```
