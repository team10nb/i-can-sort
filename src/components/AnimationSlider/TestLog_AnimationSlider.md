reason/description里面写为什么test fail了
对**组件代码**做修改和修改的理由写在每个代码块前的注释里（由于没有通过test，而对**原码**做的修改）
有必要的话，贴一下test的代码
第一个version可以不写注释 最后一个version(pass的version)可以不写reason

#### ComponentName: AnimationSlider

|Test No.|Date|Pass/Fail|Failed test(s)|Reason/Description|Modified version|
|-------|----|---------|-------------------|----------------|----------------|
|1 | 1.29 | Fail | All | No code has been constructed |version 1|
|2 | 1.29 | Pass |/|Initial version is created|version 2|
|3 | 2.5 | Fail |should match snapshot|Styles changed|version 3|
|4 | 2.5 | Pass | /                     |snapshot updated|version 4|

##### Code - version 2
```javascript
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Slider from "@material-ui/core/Slider";




export default function DiscreteSlider(props) {
    const {width, step, max, handleChange, value} = props;

    const useStyles = makeStyles({
        root: {
            width: width,
        },
    });
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <Slider
                value={typeof value === 'number' ? value : 0}
                aria-labelledby='Progress'
                valueLabelDisplay='auto'
                step={step}
                marks
                min={0}
                max={max}
                onChange={handleChange}
            />
        </div>
    );
}

```

##### Code - version 4
```javascript
import React from "react";
import { makeStyles, withStyles } from "@material-ui/core/styles";
import Slider from "@material-ui/core/Slider";

const PrettoSlider = withStyles({
    root: {
        color: "#52af77",
        height: 4,
        alignContent: "center",
    },
    thumb: {
        height: 24,
        width: 24,
        backgroundColor: "#fff",
        border: "2px solid currentColor",
        marginTop: -8,
        marginLeft: -12,
        "&:focus, &:hover, &$active": {
            boxShadow: "inherit",
        },
    },
    active: {},
    valueLabel: {
        left: "calc(-50% + 4px)",
    },
    track: {
        height: 5,
        borderRadius: 2,
    },
    rail: {
        height: 5,
        borderRadius: 2,
    },
    mark: {
        height: 5,
    }
})(Slider);

export default function DiscreteSlider(props) {
    const { step, max, handleChange, value, width } = props;

    const useStyles = makeStyles({
        root: {
            display: "grid",
            alignContent: "center",
            width: width,
            marginTop: "10px",
        },
    });
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <PrettoSlider
                value={typeof value === "number" ? value : 0}
                aria-labelledby='Progress'
                valueLabelDisplay='auto'
                step={step}
                marks
                min={0}
                max={max}
                onChange={handleChange}
            />
        </div>
    );
}


```


