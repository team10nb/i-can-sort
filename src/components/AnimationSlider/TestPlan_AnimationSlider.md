## Test Plan
--------
<br>

#### Component: AnimationSlider

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should match Snapshot | Compare new snapshot with old one. |  | Snapshots are the same. |

