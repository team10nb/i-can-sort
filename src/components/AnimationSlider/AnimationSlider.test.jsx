import { render } from '@testing-library/react';
import AnimationSlider from './AnimationSlider';

it('should match snapshot', () => {
    const mockHandleChange = jest.fn();

    const props = {
        step: 1,
        max: 10,
        handleChange: mockHandleChange,
        value: 5,
    };

    const content = render(<AnimationSlider {...props} />);
    expect(content).toMatchSnapshot();
})