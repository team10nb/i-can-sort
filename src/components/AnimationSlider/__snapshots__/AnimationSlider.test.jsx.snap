// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <span
          class="MuiSlider-root WithStyles(ForwardRef(Slider))-root-2 MuiSlider-colorPrimary"
        >
          <span
            class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-7"
          />
          <span
            class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-6"
            style="left: 0%; width: 50%;"
          />
          <input
            type="hidden"
            value="5"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
            data-index="0"
            style="left: 0%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
            data-index="1"
            style="left: 10%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
            data-index="2"
            style="left: 20%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
            data-index="3"
            style="left: 30%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
            data-index="4"
            style="left: 40%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
            data-index="5"
            style="left: 50%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
            data-index="6"
            style="left: 60%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
            data-index="7"
            style="left: 70%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
            data-index="8"
            style="left: 80%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
            data-index="9"
            style="left: 90%;"
          />
          <span
            class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
            data-index="10"
            style="left: 100%;"
          />
          <span
            aria-labelledby="Progress"
            aria-orientation="horizontal"
            aria-valuemax="10"
            aria-valuemin="0"
            aria-valuenow="5"
            class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-3 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-9"
            data-index="0"
            role="slider"
            style="left: 50%;"
            tabindex="0"
          >
            <span
              class="PrivateValueLabel-offset-11 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-5"
            >
              <span
                class="PrivateValueLabel-circle-12"
              >
                <span
                  class="PrivateValueLabel-label-13"
                >
                  5
                </span>
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <span
        class="MuiSlider-root WithStyles(ForwardRef(Slider))-root-2 MuiSlider-colorPrimary"
      >
        <span
          class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-7"
        />
        <span
          class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-6"
          style="left: 0%; width: 50%;"
        />
        <input
          type="hidden"
          value="5"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
          data-index="0"
          style="left: 0%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
          data-index="1"
          style="left: 10%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
          data-index="2"
          style="left: 20%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
          data-index="3"
          style="left: 30%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
          data-index="4"
          style="left: 40%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8 MuiSlider-markActive"
          data-index="5"
          style="left: 50%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
          data-index="6"
          style="left: 60%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
          data-index="7"
          style="left: 70%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
          data-index="8"
          style="left: 80%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
          data-index="9"
          style="left: 90%;"
        />
        <span
          class="MuiSlider-mark WithStyles(ForwardRef(Slider))-mark-8"
          data-index="10"
          style="left: 100%;"
        />
        <span
          aria-labelledby="Progress"
          aria-orientation="horizontal"
          aria-valuemax="10"
          aria-valuemin="0"
          aria-valuenow="5"
          class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-3 MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-9"
          data-index="0"
          role="slider"
          style="left: 50%;"
          tabindex="0"
        >
          <span
            class="PrivateValueLabel-offset-11 MuiSlider-valueLabel WithStyles(ForwardRef(Slider))-valueLabel-5"
          >
            <span
              class="PrivateValueLabel-circle-12"
            >
              <span
                class="PrivateValueLabel-label-13"
              >
                5
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
